boa/src/builtins/value/mod.rs

```diff
@@ -149,6 +150,96 @@ impl Value {
 
         Self::object(object)
     }
+
+    /// Convert from a JSON value to a JS value
+    pub fn from_json(json: JSONValue, interpreter: &mut Interpreter) -> Self {
+        match json {
+            JSONValue::Number(v) => {
+                Self::rational(v.as_f64().expect("Could not convert value to f64"))
+            }
+            JSONValue::String(v) => Self::string(v),
+            JSONValue::Bool(v) => Self::boolean(v),
+            JSONValue::Array(vs) => {
+                let mut new_obj = Object::default();
```

Comment:
Array should have the `prototype` too.

Reply:
Correct, I haven't done arrays, numbers, booleans, strings or null. I was able to get and set the prototype from the production code, but wasn't able to use the assertion similar to the one that I have now for the test code. As soon as I see some failing tests for the right reasons I'll address the rest of the JSON types.

Reply:
> Correct, I haven't done arrays, numbers, booleans, strings or null. I was able to get and set the prototype from the production code, but wasn't able to use the assertion similar to the one that I have now for the test code. As soon as I see some failing tests for the right reasons I'll address the rest of the JSON types.

Maybe we should make this into a separate PR.

Reply:
I'm okay with that. In fact, I was going to ask you to edit the issue with a bullet list for each JSON type.

Reply:
@Razican please take note of this thread. I noticed that you linked the PR to close the issue. I had modified the language in the PR description so that the merge would not close the issue. Please advise on whether or not this is acceptable practice for future reference.

Reply:
Yes, it's a good practise, sorry for the trouble. I have removed it.